File: java/string-2/catDog.java
/* Return true if the string "cat" and "dog" appear the same number of times 
 * in the given string.
 */
public boolean catDog(String str) {
    int cat = 0;
    int dog = 0;
      
    for(int i = 0; i < str.length() - 2; i++) {
        if(str.substring(i, i + 3).equals("cat"))
            cat++;
                        
        if(str.substring(i, i + 3).equals("dog"))
            dog++;
    }
                                      
    return cat == dog;
}

File: python/string-2/cat_dog.py
# Return True if the string "cat" and "dog" appear the same number of times 
# in the given string.
def cat_dog(str):
    cat = 0
    dog = 0
      
    for i in range(len(str) - 2):
        if str[i:i+3] == "cat":
            cat += 1
        elif str[i:i+3] == "dog":
            dog += 1
                                  
    return cat == dog
